src/nlp/equality-constrained.jl

```diff
@@ -49,6 +49,19 @@ function equality_constrained_nlp_set()
   ]
 end
 
+"""
+    eqn_solution_check(nlp, sol)
+
+Given an NLPModels `nlp` and a vector `sol`, it returns the KKT residual of an equality constrained optimization problem.
+"""
+function eqn_solution_check(nlp, sol)
+  λ = pinv(jac(nlp, sol)') * grad(nlp, sol)
```

Comment:
That's going to be fun for large sparse problems :-).

A better way to do it would be to solve an LP for λ. You could use RipQP, which is quite reliable now.

Reply:
This is supposed to be used with the basic tests created with ADNLPModels: https://github.com/JuliaSmoothOptimizers/SolverTest.jl/blob/ab87c0e28938ee47a2e66ad7a1f38d084e0a84ac/src/nlp/equality-constrained.jl#L3